lms/static/scripts/frontend_apps/components/URLFormWithPreview.tsx

```diff
@@ -21,8 +21,8 @@ export type ThumbnailData = {
 export type URLFormWithPreviewProps = {
   /** Optional extra content to be rendered together with the input and thumbnail */
   children?: ComponentChildren;
-  /** An error message to highlight that something went wrong */
-  error?: string;
+  /** A list of error messages to highlight that something went wrong */
+  errors?: string[];
```

Comment:
```suggestion
  /** An error message to highlight something went wrong */
  error?: ComponentChildren;
```

I think `URLFormWithPreview` is less cumbersome if it's not as smart about the errors, and instead it's `YouTubePicker`'s job to format the error message.